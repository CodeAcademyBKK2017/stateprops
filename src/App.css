.Cell {
    margin:5px;
    border:0;
    border-radius:5px;
    width:70px;
    display:inline-block;
    /* color:white; */
    font-size:50px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  }
.DDD{
    display:inline-block;
    text-align: center;
    font-size:100px;
    margin:5px;
}
.App {
    text-align: center;
  }
  
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
  }
  
  .App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
  }
  
  .App-title {
    font-size: 1.5em;
  }
  
  .App-intro {
    font-size: large;
  }
  
  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }