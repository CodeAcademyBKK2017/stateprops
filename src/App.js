import React, { Component } from 'react';
import './App.css';

import Counter from './components/Counter/Counter.component';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Counter />
      </div>
    );
  }
}

export default App;
