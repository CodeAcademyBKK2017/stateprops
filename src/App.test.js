import React from 'react';
import renderer from 'react-test-renderer';
import  {shallow} from 'enzyme';
import App from './App';

describe('App component', () => {
  it('App: renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('add : should increment state',() => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    expect(instance.state.value).toBe(0);
    instance.add();
    expect(instance.state.value).toBe(1);
  });

  it('subtract : should decrease state',() => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    expect(instance.state.value).toBe(0);
    instance.subtract();
    expect(instance.state.value).toBe(-1);
  });
});