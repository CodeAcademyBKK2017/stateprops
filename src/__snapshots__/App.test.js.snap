// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Overlay: renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <input
        className="Counter-box"
        type="text"
        value={0}
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "90px",
          "textAlign": "center",
        }
      }
    >
      <button
        className="Counter-button Counter-button--yellow"
        onClick={[Function]}
      >
        ADD
      </button>
      <button
        className="Counter-button Counter-button--yellow"
        onClick={[Function]}
      >
        SUBTRACT
      </button>
    </div>
  </div>
</div>
`;
